packages/smart-contracts/deploy/utils.ts

```diff
@@ -0,0 +1,173 @@
+import { Provider, Wallet, Contract } from 'zksync-web3';
+import * as hre from 'hardhat';
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+import { formatEther } from 'ethers/lib/utils';
+import { BigNumberish } from 'ethers';
+
+import '@matterlabs/hardhat-zksync-node/dist/type-extensions';
+import '@matterlabs/hardhat-zksync-verify/dist/src/type-extensions';
+import { config } from 'dotenv';
+
+config();
+
+const accounts = process.env.DEPLOYMENT_PRIVATE_KEY
+  ? [process.env.DEPLOYMENT_PRIVATE_KEY]
+  : process.env.DEPLOYER_MASTER_KEY
+  ? [process.env.DEPLOYER_MASTER_KEY]
+  : process.env.ADMIN_PRIVATE_KEY
+  ? [process.env.ADMIN_PRIVATE_KEY]
+  : undefined;
+
+const WALLET_PRIVATE_KEY = (accounts || [])[0];
+
+export const getProvider = () => {
+  const rpcUrl = hre.network.config.url;
+  if (!rpcUrl)
+    throw `⛔️ RPC URL wasn't found in "${hre.network.name}"! Please add a "url" field to the network config in hardhat.config.ts`;
+
+  // Initialize zkSync Provider
+  const provider = new Provider(rpcUrl);
```

Comment:
Is this new utils file only for zksync? We could make it apparent in the name, wdyt?

Reply:
Yes, good idea